power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -711,14 +892,27 @@ class TapPositionOptimizerImpl<std::tuple<TransformerTypes...>, StateCalculator,
                 }
                 update_state(update_data);
                 result = calculate_(state, method);
+                ++total_iterations;
             }
         }
 
         return result;
     }
 
     bool adjust_transformer(RegulatedTransformer const& regulator, State const& state, ResultType const& solver_output,
-                            UpdateBuffer& update_data) const {
+                            UpdateBuffer& update_data, SearchMethod search, BinarySearchOptions const& options) {
+        switch (search) {
+        case SearchMethod::binary_search:
+            return adjust_transformer_bs(regulator, state, solver_output, update_data, options);
+        case SearchMethod::linear_search:
+            return adjust_transformer_scan(regulator, state, solver_output, update_data);
+        default:
+            throw MissingCaseForEnumError{"TapPositionOptimizer::adjust_transformer", search};
+        }
+    }
+
+    bool adjust_transformer_scan(RegulatedTransformer const& regulator, State const& state,
```

Comment:
It might be necessary to update the method name to `adjust_transformer_linear`. Although I would prefer `scan` as it carries more accurate meaning.